policy_definitions/network_security/policy_definition_qby_VnetNamingConform.json

```diff
@@ -0,0 +1,161 @@
+{
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "name": "QBY-Deny-Vnet-Subnet-NSG",
+  "properties": {
+    "displayName": "Vnet Subnet NSG Convention",
+    "description": "Test Vnet and Subnet For Naming Convention, and if the subnet has a rule that denys all inbound traffic. Checks if the subnet has the correct NSG attached to it.",
+    "policyType": "Custom",
+    "mode": "Indexed",
+    "metadata": {
+      "version": "1.0.0",
+      "category": "NetworkSecurity"
+    },
+    "parameters": {
+      "effect": {
+        "type": "String",
+        "defaultValue": "Deny",
+        "allowedValues": [
+          "Audit",
+          "Deny",
+          "Disabled"
+        ],
+        "metadata": {
+          "displayName": "Effect",
+          "description": "The effect determines what happens when the policy rule is evaluated to match"
+        }
+      }
+    },
+    "policyRule": {
+      "if": {
+        "allOf": [
+          {
+            "field": "type",
+            "equals": "Microsoft.Network/virtualNetworks"
```

Comment:
THis policy only checks Subnets, that are created/changed together with the vNET. If you just add a vNET, this policy is most likely ignored. This should be noted somewhere.

Reply:
You added a policy, that takes care of not inline created subnets